test/models/user_test.rb

```diff
@@ -77,6 +77,79 @@ class UserTest < ActiveSupport::TestCase
     assert_equal sponsorer_detail_2, user.sponsorer_detail
   end
 
+  test 'append @ prefix in twitter handle if not present when created' do
+    user = create :user
+    assert_nil user.twitter_handle
+    user.set(twitter_handle: 'amitk301293')
+    assert "@amitk301293", user.twitter_handle
+  end
+
+  test 'twitter handle must not be blank when created' do
+    user = create :user
+    assert_nil user.twitter_handle
+    user.reload.set(twitter_handle: "")
+    assert_empty user.reload.twitter_handle
+    assert_not user.valid?
+    user.reload.set(twitter_handle: "@amik301293")
+    assert user.valid?
+  end
+
+  test 'twitter handle must not contain spaces when created' do
+    user = create :user
+    user.set(twitter_handle: "@amitk 301293")
+    assert_not user.valid?
+  end
+
+  test 'twitter handle must not contain special characters when created' do
+    user = create :user
+    user.set(twitter_handle: "@amit@k#301293")
+    assert_not user.valid?
+  end
+
+  test 'twitter handle length must be 15 or less when created' do
+    user = create :user
+    user.set(twitter_handle: "@aswisdakimasdfeassdfsasdfsere")
+    assert_not user.valid?
+  end
+
+  test 'append @ prefix in twitter handle if not present when updated' do
+    user = create :user
+    assert_nil user.twitter_handle
+    user.update(twitter_handle: 'amitk301293')
+    assert "@amitk301293", user.twitter_handle
+  end
+
+  test 'twitter handle must not be blank when updated' do
+    user = create :user
+    assert_nil user.twitter_handle
+    user.reload.update(twitter_handle: "")
+    assert_empty user.twitter_handle
+    assert_not user.valid?
+    user.reload.set(twitter_handle: "@amik301293")
+    assert user.valid?
+  end
+
+  test 'twitter handle must not contain spaces when updated' do
+    user = create :user
+    user.set(twitter_handle: "@amitk")
+    assert user.valid?
+    user.update(twitter_handle: "@amitk 301293")
+    assert_not user.valid?
+  end
+
+  test 'twitter handle must not contain special characters when updated' do
+    user = create :user
+    user.set(twitter_handle: "@amitk301293")
+    user.update(twitter_handle: "@amitk#301293")
+    assert_not user.valid?
+  end
+
+  test 'twitter handle length must be 15 or less when updated' do
+    user = create :user
+    user.update(twitter_handle: "@aswisdakimasdfeassdfsasdfsere")
+    assert_not user.valid?
+  end
+
```

Comment:
you are already creating the user and then setting its value(i.e updating it). hence, the '..when user is created' test cases arent necessary.